contributors/design-proposals/pod-priority-api.md

```diff
@@ -0,0 +1,203 @@
+may have other restrictions on preempting pods, for example, it may refuse to
+preempt a pod if PodDisruptionBudget is violated. The details of scheduling and
+preemption decisions are beyond the scope of this document.
+
+### Priority in PodSpec 
+
+Pods may have priority in their pod spec. PodSpec has field called
+"PriorityName" which is specified by user. PodStatus has a corresponding field
+called "Priority". User-specified priority (PriorityName) is a string and all
+of the valid priority names are defined by a system wide mapping that maps each
+string to an integer. The PriorityName specified in a pod spec must be found in
+this map or the pod creation request will be rejected. If PriorityName is
+empty, it will resolve to priority zero. Once the PriorityName is resolved to an
+integer, it is placed in "Priority" field of PodStatus.
+
+
+```
+type PodSpec struct {
+  ...
+  PriorityName string
+}
+
+type PodStatus struct {
+  ...
+  Priority int  // Populated by Admission Controller.
+}
+```
+
+### Priority names 
+
+A Kubernetes cluster has four predefined (built-in) priority names. The top
+priority is reserved for the system pods and its value is the largest 32-bit
+integer. The following list is an example of how the priorities and their
+values may look like. Kubernetes will not be shipped with predefined priority
+names, except for the "system" priority which will be built in Kubernetes.
+The decision to ship with no other priority names was made to avoid creating
+defacto standard names which may be hard to change in the future.
+
+
+```
+system  2147483647 (int_max)
+tier1   4000
+tier2   2000
+tier3   1000
+```
+
+The following shows a list of example workloads in a Kubernetes cluster in decreasing order of priority:
+
+* Kubernetes system daemons (per-node like fluentd, and cluster-level like 
+  Heapster)
+* Critical user infrastructure (e.g. storage servers, monitoring system like
+  Prometheus, etc.)
+* Components that are in the user-facing request serving path and must be able 
+  to scale up arbitrarily in response to load spikes (web servers, middleware,
+  etc.)
+* Important interruptible workloads that need strong guarantee of
+  schedulability and of not being interrupted
+* Less important interruptible workloads that need a less strong guarantee of
+  schedulability and of not being interrupted
+* Best effort / opportunistic
+
+### Resolving priority names 
+
+User requests sent to Kubernetes may have "PriorityName"  in their PodSpec.
+Admission controller resolves a PriorityName to its corresponding number and
+populates the "Priority" field of the pod status. The rest of Kubernetes
+components look at the "Priority" field of pod status and work with the integer
+value. In other words, "PriorityName" will be ignored by the rest of the
+system.
+
+### Modifying priority names 
+
+Admission controller can optionally receive a ConfigMap from the API server
```

Comment:
I mentioned my above comment to @thockin and he pointed out it doesn't really work since ResourceQuota is per-namespace and we want the mapping from priority name to priority to be global, not per-namespace.

However, if we were to upstream the OpenShift [cross-namespace quota](https://docs.openshift.com/container-platform/3.5/admin_guide/multiproject_quota.html) then we could put the mapping in that object.